### ISSUE TITLE
Role "Button" not announced by TalkBack for Spans with GestureRecognizer

### ISSUE THREAD

**Issue by ChristopherStephan** [dotnet/maui#25763]
### Description

When using a formatted label with a hyperlink, Android's screen reader TalkBack does not announce the role "Button" for the hyperlink.

```
<Label>
    <Label.FormattedText>
        <FormattedString>
            <Span Text="Alternatively, click " />
            <Span Text="here"
                  TextColor="Blue"
                  TextDecorations="Underline">
                <Span.GestureRecognizers>
                    <TapGestureRecognizer Command="{Binding TapCommand}"
                                          CommandParameter="https://learn.microsoft.com/dotnet/maui/" />
                </Span.GestureRecognizers>
            </Span>
            <Span Text=" to view .NET MAUI documentation." />
        </FormattedString>
    </Label.FormattedText>
</Label>
```

Actual outcome: "Alternatively, click here to view .NET MAUI documentation"
Expected outcome:  "Alternatively, click here to view .NET MAUI documentation., Button"

On iOS with VoiceOver the button role is announced as expected.

### Steps to Reproduce

1. Create a Label with a Span that has a Gesture Recognizer
2. Start Android's screen reader TalkBack

Actual outcome: The screen reader does not announce the role "Button" of the Span that has a Gesture Recognizer
Expected outcome: The role "Button" is announced

### Version with bug

8.0.40 SR5

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Affected platforms

Android

### Relevant log output

```shell

```

**Comment by ninachen03** [comment]
This issue has been verified using Visual Studio 17.12.0 Preview 5(9.0.0-rc.2.24503.2 & 8.0.92 & 8.0.3). Can repro this issue at android platform

**Comment by mattleibow** [comment]
@ninachen03 do you still have the repro you created?

**Comment by ninachen03** [comment]
@mattleibow here is repro for you.
 [MauiApp37.zip](https://github.com/user-attachments/files/17713697/MauiApp37.zip)
